src/Handler/Table/Create/CreateTableHandler.php

```diff
@@ -0,0 +1,87 @@
+use Keboola\StorageDriver\Command\Table\CreateTableResponse;
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+use Keboola\StorageDriver\Shared\Driver\Exception\Exception;
+use Keboola\StorageDriver\Teradata\ConnectionFactory;
+use Keboola\TableBackendUtils\Column\ColumnCollection;
+use Keboola\TableBackendUtils\Column\Teradata\TeradataColumn;
+use Keboola\TableBackendUtils\Escaping\Teradata\TeradataQuote;
+use Keboola\TableBackendUtils\Table\Teradata\TeradataTableQueryBuilder;
+
+final class CreateTableHandler implements DriverCommandHandlerInterface
+{
+    /**
+     * @inheritDoc
+     * @param GenericBackendCredentials $credentials
+     */
+    public function __invoke(
+        Message $credentials,
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof CreateTableCommand);
+
+        $db = ConnectionFactory::getConnection($credentials);
+
+        // root user is also root database
+        $databaseName = $credentials->getPrincipal();
+        $meta = $command->getMeta();
+        if ($meta !== null) {
+            // override root user and use other database as root
+            $meta = $meta->unpack();
+            assert($meta instanceof CreateProjectCommand\CreateProjectTeradataMeta);
+            $databaseName = $meta->getRootDatabase() === '' ? $databaseName : $meta->getRootDatabase();
+        }
+
+        // columns
+        $columns = [];
+        /** @var TableColumn $column */
+        foreach ($command->getColumns() as $column) {
+            $columnDefinition = new Teradata($column->getType(), [
+                'length' => $column->getLength(),
+                'nullable' => $column->getNullable(),
+                'default' => $column->getDefault(),
+                // TODO isLatin
+            ]);
+            $columns[] = new TeradataColumn($column->getName(), $columnDefinition);
+        }
+        $columnsCollection = new ColumnCollection($columns);
+
+        // build sql
+        $builder = new TeradataTableQueryBuilder();
+        $createTableSql = $builder->getCreateTableCommand($databaseName, $command->getTableName(), $columnsCollection, $command->getPrimaryKeysNames());
+
+        // create table
+        $db->executeStatement($createTableSql);
+
+        // check table exits
+        $tables = $db->fetchAllAssociative(sprintf(
+            'SELECT TableName FROM DBC.TablesX WHERE TableName = %s',
+            TeradataQuote::quote($command->getTableName())
+        ));
+        if (count($tables) === 0) {
+            throw new Exception(sprintf(
+                'Table "%s" should exists in database "%s" for user "%s".',
+                $command->getTableName(),
+                $databaseName,
+                $credentials->getPrincipal()
+            ));
+        }
```

Comment:
Tady to nepotřebuješ věříme, že to tabulku vytvoří tak jak chceme když to nespadne, je to otestované v utils a měl by to asi tady otestovat taky funkcionální test.

Reply:
ok, smazano